### ISSUE TITLE
Don't make curl with pipe failed by command-instead-of-module: curl used in place of get_url or uri module

### ISSUE THREAD

**Issue by abcfy2** [ansible/ansible-lint#2677]
##### Summary

Sometimes I really need to use curl for my script:

```yml
- name: "Add nginx repo keyring"
  ansible.builtin.shell: curl -fsSL https://nginx.org/keys/nginx_signing.key | gpg --dearmor > /etc/apt/keyrings/nginx-archive-keyring.gpg
```

Failed with:

```sh
command-instead-of-module: curl used in place of get_url or uri module
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```
command-instead-of-module: curl used in place of get_url or uri module
```

```console (paste below)
ansible --version
ansible [core 2.14.0]
  config file = /home/fengyu/projects/devops/ansible_playbooks/ansible.cfg
  configured module search path = ['/home/fengyu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/fengyu/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0] (/usr/bin/python)
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.8.6 using ansible 2.14.0
```

- ansible installation method: one of source, pip, OS package

- ansible-lint installation method: one of source, pip, OS package

##### OS / ENVIRONMENT
Manjaro Linux

##### STEPS TO REPRODUCE

```yml
- name: "Add nginx repo keyring"
  ansible.builtin.shell: curl -fsSL https://nginx.org/keys/nginx_signing.key | gpg --dearmor > /etc/apt/keyrings/nginx-archive-keyring.gpg
```

```console (paste below)
command-instead-of-module: curl used in place of get_url or uri module
```

**Comment by ssbarnea** [comment]
Sometimes reading documentation can provide solutions, https://ansible-lint.readthedocs.io/rules/command-instead-of-module/
